aspnetcore/signalr/troubleshoot.md

```diff
@@ -0,0 +1,88 @@
+---
+title: ASP.NET Core SignalR connection troubleshooting
+author: 
+description: ASP.NET Core SignalR connection troubleshooting.
+monikerRange: '>= aspnetcore-2.1'
+ms.author: 
+ms.custom: mvc
+ms.date: 04/08/2020
+no-loc: ["ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: signalr/troubleshoot
+---
+# Troubleshoot connection errors
+
+This section provides help with errors that can occur when trying to establish a connection to a ASP.NET Core SignalR hub.
+
+### Response code 404
+
+When using WebSockets and `skipNegotiation = true`
+```log
+WebSocket connection to 'wss://xxx/HubName' failed: Error during WebSocket handshake: Unexpected response code: 404
+```
+
+* If the connection takes too long to send a request to the server after the negotiate the server will delete the ID and return a 404 if the connection uses that ID
```

Comment:
This is difficult for native speakers to read and impossible to MT (machine translate)

```suggestion
* If the connection takes too long to send a request to the server after the negotiate, the server:

  * Deletes the ID.
  * Returns a 404 if the connection uses that ID.
```

Reply:
Does it only return 404 if the connection uses that ID? Maybe the following is better:
```suggestion
* If the connection uses the ID and takes too long to send a request to the server after the negotiate, the server:

  * Deletes the ID.
  * Returns a 404.
```